assets/src/edit-story/components/panels/stylePreset/presets.js

```diff
@@ -0,0 +1,279 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import { useState, useRef, useLayoutEffect } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ReactComponent as Remove } from '../../../icons/remove.svg';
+import { BACKGROUND_TEXT_MODE } from '../../../constants';
+import generatePatternStyles from '../../../utils/generatePatternStyles';
+import { PanelContent } from '../panel';
+import { useKeyDownEffect } from '../../keyboard';
+import { generatePresetStyle } from './utils';
+
+const PRESET_HEIGHT = 35;
+const REMOVE_ICON_SIZE = 18;
+const COLORS_PER_ROW = 6;
+const STYLES_PER_ROW = 3;
+
+const presetCSS = css`
+  display: inline-block;
+  width: 30px;
+  height: 30px;
+  border-radius: 15px;
+  border-color: transparent;
+  padding: 0;
+  font-size: 13px;
+  position: relative;
+  svg {
+    width: ${REMOVE_ICON_SIZE}px;
+    height: ${REMOVE_ICON_SIZE}px;
+    position: absolute;
+    top: calc(50% - ${REMOVE_ICON_SIZE / 2}px);
+    left: calc(50% - ${REMOVE_ICON_SIZE / 2}px);
+  }
+`;
+
+const Color = styled.button`
+  ${presetCSS}
+  ${({ color }) => generatePatternStyles(color)}
+`;
+
+const Style = styled.button`
+  ${presetCSS}
+  background: transparent;
+  ${({ styles }) => styles}
+  width: 72px;
+  border-radius: 4px;
+`;
+
+const PresetGroup = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+`;
+
+const ButtonWrapper = styled.div`
+  flex-basis: ${100 / COLORS_PER_ROW}%;
+  height: ${PRESET_HEIGHT}px;
+`;
+
+const StyleButtonWrapper = styled.div`
+  flex-basis: ${100 / STYLES_PER_ROW}%;
+  height: ${PRESET_HEIGHT}px;
+`;
+
+const PresetGroupLabel = styled.div`
+  color: ${({ theme }) => theme.colors.fg.v1};
+  font-size: 10px;
+  line-height: 12px;
+  text-transform: uppercase;
+  padding: 6px 0;
+`;
+
+const TextWrapper = styled.span`
+  text-align: left;
+  line-height: 1.3;
+`;
+
+const HighLight = styled.span`
+  padding: 0 2px;
+  ${({ background }) => generatePatternStyles(background)}
+  box-decoration-break: clone;
+`;
+
+function Presets({
+  stylePresets,
+  getEventHandlers,
+  isEditMode,
+  isText,
+  textContent = 'Text',
+}) {
+  const [activeColorIndex, setActiveColorIndex] = useState(null);
+  const [activeStyleIndex, setActiveStyleIndex] = useState(null);
+  const colorsRef = useRef(null);
+  const stylesRef = useRef(null);
+  const { fillColors, textColors, textStyles } = stylePresets;
+
+  const getIndexDiff = (key, rowLength) => {
+    switch (key) {
+      case 'ArrowUp':
+        return -rowLength;
```

Comment:
Why is this not just `-COLORS_PER_ROW`?

Reply:
(or the other way around for down)

Reply:
There's also `STYLES_PER_ROW` which is different.

Reply:
The `COLORS_PER_ROW` below is actually a mistake, just realized now.